src/main/java/supersymmetry/mixins/gregtech/MultiblockMachineMixin.java

```diff
@@ -0,0 +1,31 @@
+package supersymmetry.mixins.gregtech;
+
+import gregtech.api.metatileentity.multiblock.MultiblockWithDisplayBase;
+import gregtech.common.metatileentities.multi.electric.MetaTileEntityFluidDrill;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.spongepowered.asm.mixin.Mixin;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Mixin(MetaTileEntityFluidDrill.class)
+public abstract class MultiblockMachineMixin extends MultiblockWithDisplayBase {
+
+    public MultiblockMachineMixin(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId);
+    }
+
+    @Override
+    public boolean getIsWeatherOrTerrainResistant() {
+        return true;
+    }
+
+    public void addInformation(ItemStack stack, @Nullable World player, @NotNull List<String> tooltip, boolean advanced) {
```

Comment:
what about other multis in ceu
I mean it would look a bit strange if drilling rig is the only machine that has this tooltip

also as a note overriding this would remove all tooltips defined in `MetaTileEntityFluidDrill.class` itself

You can probably add a mixin targeting MultiblockBase.class, and check `getIsWeatherOrTerrainResistant()`, if returns `true`, then do `tooltip.add(I18n.format("susy.weather_resistance"))`

Reply:
>other multis in ceu
Only outdoors machines count, so fluid drill, ore drill rig, singleblock pumps/miners.

Also the code would probably better if the tooltip is added for_every machines_ having getIsWeatherOrTerrainResistant() = true.

Reply:
> > other multis in ceu
> 
> Only outdoors machines count, so fluid drill, ore drill rig, singleblock pumps/miners.
> 
> Also the code would probably better if the tooltip is added for _every machine_ which has getIsWeatherOrTerrainResistant() = true.

that's indeed the case. That method is in the very basic class of machines (MetaTileEntity,class actually)

Reply:
so if you want to do with single blocks, just target MetaTileEntity.class